### ISSUE TITLE
[BUG] floorplan.text_set wrap the output text inside a <title/> tag

### ISSUE THREAD

**Issue by MarlBurroW** [ExperienceLovelace/ha-floorplan#143]
**Describe the bug**
When I use `floorplan.text_set` to display the state of an entity, the output text is wrapped inside a <title> tag.
Text value is not visible because a title in a text tag is probably not a valid SVG semantic.

**To Reproduce**
1. Create a simple SVG with a text tag `<text id="temperature"></text> inside`
2. Add a rule to set the text tag with an entity state.
```yaml
rules:
  - element: temperature
    entity: sensor.my_temperature
    state_action:
      - service: floorplan.text_set
        service_data: '${entity.state ? entity.state + "°C" : "unknown"}'
```

**Expected behavior**

```html
<text id="temperature">29°C</text>
```

**Observed behavior**
```html
<text id="temperature"><title>29°C</title></text>
```

**Screenshots**
![alt text](https://d3dehtdmp2rwcw.cloudfront.net/ms_254763/HXUJC2RWTUhOXm0f6kJngx206eOLFQ/Floorplan%2B-%2BHome%2BAssistant%2B2021-09-05%2B00-49-20.png?Expires=1630796400&Signature=UaLs7sM8Yc4Ai4gGZvx-yu83Vs-JOTYH4IprzdVh4lEzx0dai7HQXJMGje7OKD10m-GZcH-IJE4B3ZIP4a6Xm3~7TbwgqBQvfuyjAPt6tPBnBLBveWxCOouqhqYDq3hRRna8fXFd0JVg050QPB72nYuo3pijB9qB5z1H~MTmxmdqwlPhd48AUtiUmOdWyzku65VmV5ZynXNG0l9qJsfC~Nkof7~NqE7cvaNQ-HkWMADQeho-1JMhczbisTz5XMZ7N7~ajtft5TxPmMsNRjK~U13vXrrCGnepdaDepcHZyR9CJmVXqzzATCKjYysOCGE2IJ491GZkkKB9Vg~23I7Ukw__&Key-Pair-Id=APKAJBCGYQYURKHBGCOA "Logo Title Text 1")

**Desktop (please complete the following information):**
 - OS: MacOSX 11.2.2
 - Browser: Chrome 93.0.4577.63 (Official Build) (x86_64)
 - HA: 2021.9.1 (HAOS x86 on a NUC)
 - HA Frontend: 20210830.0
 - ha-floorplan: 1.0.27

**Comment by OzGav** [comment]
I don't have any problems with `text_set`. The only issue I had was setting up initially in inkscape. Make sure you select the text you want to replace and then select TEXT>>CONVERT TO TEXT

**Comment by MarlBurroW** [comment]
Ok understood:

I've build my SVG with Illustrator. Illustrator generate only a `<text/>` tag when you add a text area.
I've downloaded Inkscape to see the output, and it add a `<tspan>` inside the `<text/>` tag.

So I have manually added a `<tspan/>` inside all my `<text/>` tags and it works:

![image](https://user-images.githubusercontent.com/3024430/132371973-92afb843-208e-4c67-9908-2c0644a7dd52.png)

Ha-Floorplan seems to automatically select the child tspan of the given element to insert the `text_set`. Maybe it deserves to be specified in the doc.

**Comment by exetico** [comment]
I totally agree, and have already asked @pkozul about the current logics. It's clear that we're forcing the titel-tag, and I need to know how inportant it actually is 😊👍. 

Due to lag of time, I've not found time to run through the related logics.

I'm glad that you figured it out. And if no fix are provided, I'd definitely add that!

**Comment by MarlBurroW** [comment]
Thanks @exetico 

I made a PR to add a troubleshooting, but i'm not sure this is the best solution.

I don't know exactly why the `setText` function create a `<title>` element:

```ts
// /src/lib/utils.ts
  static setText(
    textElement: HTMLElement | SVGGraphicsElement,
    text: string
  ): void {
    const tspanElement = textElement.querySelector('tspan');
    if (tspanElement) {
      tspanElement.textContent = text;
    } else {
      let titleElement = textElement.querySelector('title') as Element;
      if (!titleElement) {
        titleElement = document.createElementNS(
          'http://www.w3.org/2000/svg',
          'title'
        );
        textElement.appendChild(titleElement);
      }
      titleElement.textContent = text;
    }
  }
```
Is it for the tooltip ? 
Why not create a tspan instead ? or directly set the `textElement.textContent = text` ?

**Comment by exetico** [comment]
That's the only reason why I've asked him 👍 We need to be alligned, before I change the structure (I had the excact same thoughts). Hopefully @petarkozul will fix it, else, we'll do something else about it 😊.

**Comment by MarlBurroW** [comment]
Ok found:

https://github.com/ExperienceLovelace/ha-floorplan/blob/2a9b564a99f200a04d33fe77e8573e4535c6371c/src/components/floorplan/floorplan-element.ts#L1508-L1536

Here, all `<title>` tags inside the target element are selected and all attributes are inserted inside.

it confirms that it is for the tooltip displaying entity attributes.

It seems to have a conflict between the `hover-info` and the `text_set` target.

Current behavior is:

- When a SVG element doesn't have a `<title/>` child tag and have a `hover-info` attached, a `<title/>` tag is created/selected and tooltip updates goes inside.
- When a SVG element doesn't have a `<tspan/>` child tag and have a `text_set` configured, a `<title/>` tag is created/selected and text updates goes inside. But If `<tpsan/>` exists, text updates goes inside.

The tooltips updates and text goes to the same `<title/>` attribute if there is no `<tspan/>` inside the `<text/>` tag.

Expected Behavior.

- When a SVG element doesn't have a title tag and have a `hover-info` attached, a `<title/>` tag should be added/selected to insert tooltip data updates.
- When a SVG element doesn't have a `<tspan/>` tag inside and have a `text_set` configured, a `<tspan/>` tag should be added inside the `<text/>` tag, and all next text updates should be inserted in this `<tspan>`

I'm not sure to be clear, but I think this can be considered as a bug.

**Comment by MarlBurroW** [comment]
I think this can be solved by just updating the `setText` function to create a tspan instead of a title:

```ts
  static setText(
    textElement: HTMLElement | SVGGraphicsElement,
    text: string
  ): void {
    let tspanElement = textElement.querySelector('tspan')

    if(!tspanElement) {
      document.createElementNS( 'http://www.w3.org/2000/svg', 'tspan');
      textElement.appendChild(tspanElement);
    }

    tspanElement.textContent = text;
  }
```
No longer need to create a <title> element in this function, because the hover-info function already add a <title> if it doesn't exists. see here: https://github.com/ExperienceLovelace/ha-floorplan/blob/2a9b564a99f200a04d33fe77e8573e4535c6371c/src/components/floorplan/floorplan-element.ts#L1508-L1536

For me it should solve the issue, but I do not know enough about the project to know what the possible negative impacts will be

**Comment by exetico** [comment]
Fixed with #144 - Thanks!
